From 87f1c81680c762bce9000cc2e8b479c91afd6837 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Thu, 18 Nov 2010 16:17:32 +0800
Subject: [PATCH] ENGR00133848 imx51 MSL uart: change uart default parent to pll2

change uart default parent to pll2 and clk rate to 66.5MHz

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 arch/arm/mach-mx5/clock.c |   11 ++++-------
 1 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-mx5/clock.c b/arch/arm/mach-mx5/clock.c
index 7463870..783d973 100644
--- a/arch/arm/mach-mx5/clock.c
+++ b/arch/arm/mach-mx5/clock.c
@@ -4660,17 +4660,14 @@ int __init mx51_clocks_init(unsigned long ckil, unsigned long osc, unsigned long
 		    (0 << MXC_CCM_CSCDR1_UART_CLK_PODF_OFFSET);
 		__raw_writel(reg, MXC_CCM_CSCDR1);
 	} else {
-		/* Move UART to run from PLL3 */
-		clk_set_parent(&uart_main_clk, &pll3_sw_clk);
+		clk_set_parent(&uart_main_clk, &pll2_sw_clk);
 
-		/* Set the UART dividers to divide,
-		 * so the UART_CLK is 66.5MHz.
-		 */
+		/* Set the UART dividers to divide, so the UART_CLK is 66.5MHz. */
 		reg = __raw_readl(MXC_CCM_CSCDR1);
 		reg &= ~MXC_CCM_CSCDR1_UART_CLK_PODF_MASK;
 		reg &= ~MXC_CCM_CSCDR1_UART_CLK_PRED_MASK;
-		reg |= (3 << MXC_CCM_CSCDR1_UART_CLK_PRED_OFFSET) |
-		    (0 << MXC_CCM_CSCDR1_UART_CLK_PODF_OFFSET);
+		reg |= (4 << MXC_CCM_CSCDR1_UART_CLK_PRED_OFFSET) |
+		    (1 << MXC_CCM_CSCDR1_UART_CLK_PODF_OFFSET);
 		__raw_writel(reg, MXC_CCM_CSCDR1);
 	}
 
-- 
1.5.4.4

